androidApp/src/main/java/com/example/tasky/android/agenda/screen/AgendaScreen.kt

```diff
@@ -48,9 +53,32 @@ import kotlinx.datetime.plus
 import kotlinx.datetime.toLocalDateTime
 import kotlinx.serialization.Serializable
 
-fun NavGraphBuilder.agendaScreen() {
+fun NavGraphBuilder.agendaScreen(
+    navigateToCreateEvent: () -> Unit,
+    navigateToCreateTask: () -> Unit,
+    navigateToCreateReminder: () -> Unit,
+    navigateToAgendaDetails: (AgendaItem) -> Unit,
+    navigateToEditAgenda: (AgendaItem) -> Unit,
+) {
     composable<Agenda> {
-        AgendaScreen(AgendaScreenState(), onEvent = {})
+        val viewModel: AgendaViewModel = viewModel()
+
+        val screenState by viewModel.screenStateFlow.collectAsStateWithLifecycle()
+
+        LaunchedEffect(true) {
+            viewModel.initState()
+        }
```

Comment:
Initializing the state from the UI has a major downside: This will be called with every configuration change like a screen rotation or changing the theme or simply whenever going back to this screen. 

While there's no objective truth about where to load initial data on Android, I would consider doing this either in the VM's `init` function (which triggers just once) or when your `screenStateFlow` collection starts. I go over the different possibilities in my video:
https://www.youtube.com/watch?v=mNKQ9dc1knI

Reply:
The UI layer (Jetpack Compose) is primarily a visual representation responsible for rendering state, so it’s odd that it is also responsible for managing data initialization. Also, LaunchedEffect may re-trigger on each initial composition; thus, if the composable's lifecycle differs from the ViewModel's, it can cause unwanted behavior.